### PR TITLE
feat: support multiple choices

### LINKED ISSUE #264: feat: [support return multiple choices]
**Problem**
- Support params: `n` `integer` or null
- Optional
- Defaults to 1
How many chat completion choices to generate for each input message. Note that you will be charged based on the number of generated tokens across all of the choices. Keep n as 1 to minimize costs.

-> need to check if llama.cpp support this option.

reference: https://platform.openai.com/docs/api-reference/chat/create#chat-create-n

related issue: https://github.com/janhq/internal/issues/160

### PR DESCRIPTION
**Pull request by nguyenhoangthuan99**
Fix #264

### CODE REVIEW

File: src/llama_engine.cc [hunk 1]
```diff
@@ -610,6 +610,7 @@ void LlamaEngine::HandleInferenceImpl(
   data["n_probs"] = completion.n_probs;
   data["min_keep"] = completion.min_keep;
   data["grammar"] = completion.grammar;
+  data["n"] = completion.n;
```
Comment (vansangpfiev) [inline src/llama_engine.cc:613]: Please add a comment for `n` variable

File: src/llama_engine.cc [hunk 2]
```diff
@@ -835,31 +836,53 @@ void LlamaEngine::HandleInferenceImpl(
...
         }
+        // TaskResult result = state->llama.NextResult(task_id);
+        for (auto& result : results) {
+          if (!result.error && result.stop) {
+            prompt_tokens += result.result_json["tokens_evaluated"].get<int>();
+            predicted_tokens +=
+                result.result_json["tokens_predicted"].get<int>();
+            std::string to_send = result.result_json["content"];
+            llama_utils::ltrim(to_send);
+            if (respData.empty()) {
```
Comment (vansangpfiev) [inline src/llama_engine.cc:867]: Does this mean we do not append the first result to `choices`?

Comment (nguyenhoangthuan99) [reply src/llama_engine.cc:867]: respData is created as empty Json::Value, if respData is empty we will directly assign result to it

File: src/llama_engine.cc [hunk 3]
```diff
@@ -835,31 +836,53 @@ void LlamaEngine::HandleInferenceImpl(
...
-      int task_id = state->llama.RequestCompletion(d, false, false, -1);
+      std::vector<int> task_ids;
+      for (int i = 0; i < n; i++) {
+        task_ids.push_back(state->llama.RequestCompletion(d, false, false, -1));
+      }
       LOG_INFO << "Request " << request_id << ": "
                << "Non stream, waiting for respone";
       if (!json_value(d, "stream", false)) {
         bool has_error = false;
         std::string completion_text;
```
Comment (vansangpfiev) [inline src/llama_engine.cc:852]: Seems like we don't use this variable. Can you help to remove it?

File: src/llama_engine.cc (hunk 2 above)
Comment (vansangpfiev) [reply src/llama_engine.cc:867]: Got it. Thank you,

**Review by vansangpfiev — APPROVED** [review]

**Pull request merged**